devlib/collector/ftrace.py

```diff
@@ -412,15 +502,15 @@ def get_data(self):
         # The size of trace.dat will depend on how long trace-cmd was running.
         # Therefore timout for the pull command must also be adjusted
         # accordingly.
-        pull_timeout = 10 * (self.stop_time - self.start_time)
+        pull_timeout: float = 10 * (cast(float, self.stop_time) - cast(float, self.start_time))
```

Comment:
A single top-level cast should be enough

Reply:
Ok so that means `self.start_time` can be `None`, which should not be the case at the point where it's valid to call that method. It's a similar problem to the attributes set in `Target.connect()`. We need to somewhat teach mypy about the expected state machine.